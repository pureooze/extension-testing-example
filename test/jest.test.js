import chrome from "./mocks.js";
import React from "react";
import ReactDOM from "react-dom";
import { configure, mount } from "enzyme";
import Main from "../src/example/url.jsx";
import { Adapter } from "enzyme-adapter-preact";

configure({ adapter: new Adapter() });

test("should render", () => {
  chrome.runtime.getURL.returns("http://localhost/popup-content.html");
  const wrapper = mount(<Main />);
  expect(wrapper.toJSON()).toMatchSnapshot();
});
