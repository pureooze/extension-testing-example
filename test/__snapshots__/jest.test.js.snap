// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
ReactWrapper {
  Symbol(enzyme.__unrendered__): <Url />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "children": Array [],
    "instance": <undefined />,
    "nodeType": "host",
    "props": Object {},
    "type": "undefined",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "children": Array [],
      "instance": <undefined />,
      "nodeType": "host",
      "props": Object {},
      "type": "undefined",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": PreactAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
